core/templates/dev/head/filtersSpec.js

```diff
@@ -180,4 +182,37 @@ describe('Testing filters', function() {
     // tested here.
     expect(filter('Single line\r\nWindows EOL')).toEqual('Single line...');
   }));
+
+  it('should summarize large number to maximum of four significant figures and append metrix prefix',
+       inject(function($filter) {
+    var filter = $filter('summarizeNonnegativeNumber');
+
+    expect(filter(100)).toEqual(100);
+    expect(filter(1720)).toEqual('1.7K');
+    expect(filter(2306200)).toEqual('2.3M');
+
+    expect(filter(12389654281)).toEqual('12.4B');
+    expect(filter(897978581123)).toEqual('898.0B');
+    expect(filter(476678)).toEqual('476.7K');
+
+  }));
+
+  it('should capitalize first letter and truncate string ', inject(function($filter) {
+    var filter = $filter('truncateAndCapitalize');
+
+    expect(filter('remove New line', 4)).toEqual('Remo...');
+    // If maximum number of character are not specified
+    // return whole objective but first letter capitalize.
```

Comment:
// If the maximum number of characters is not specified, return
// the whole objective with the first letter capitalized.